src/SurfaceFluxes.jl

```diff
@@ -357,50 +363,83 @@ function obukhov_length(
     sc::AbstractSurfaceConditions{FT},
     uft::UF.AUFT,
     scheme;
-    tol::RS.AbstractTolerance = RS.SolutionTolerance(max(FT(1e-1), sqrt(eps(FT)))),
+    tol::RS.AbstractTolerance = RS.SolutionTolerance(FT(z_in(sc) / 50)),
```

Comment:
maybe we can use `Δz(sc::AbstractSurfaceConditions) = z_in(sc) - z_sfc(sc)`? if not should we expose this value as a parameter?

Reply:
The idea is to make the tolerance dependent on dz of the first layer, so the denominator should be a fixed value rather than z_in(sc) - z_sfc(sc). `tol` is a parameter that can be changed by the user. Another choice is to keep this fixed at 0.1 and specify it in the tests, climaatmos examples, etc.

Reply:
I think that specifying `tol` I the correct thing to do, it is also non-dimensional right?

Reply:
This is the absolute tolerance for obukhov length, so it has unit of meter. It would be better to set it as a relative tolerance I think, but with the current root solver structure it’s not possible to do so.